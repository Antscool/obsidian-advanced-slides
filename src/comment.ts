export type Comment = {

	type: string,
	style: string[],
	clazz: string[],
	attributes: Map<string, string>,
}

export class CommentParser {

	private readCommentRegex = /<!--.*-->/;
	private styleRegex = /(?:style="([^"]+)")|(?:style='([^']+)')/;
	private classRegex = /(?:class="([^"]+)")|(?:class='([^']+)')/;
	private parseRegex = /<!--\s*\.(element|slide):\s*(.*)-->/;
	private parsePropertiesRegex = /([^=]*)\s*=\s*"([^"]*)"\s*|([^=]*)\s*=\s*'([^']*)'\s*/g;

	commentToString(comment: Comment): string {
		return `<!-- .${comment.type}: ${this.buildAttributes(comment)} -->`;
	}

	buildAttributes(comment: Comment): string {
		return `style="${comment.style.join('; ')}" class="${comment.clazz.join(' ')}" ${this.attributeMapToString(comment.attributes)}`;
	}

	parseLine(line: string): Comment {
		return this.lineHasComment(line)
			? this.parseComment(this.readCommentStringFromLine(line))
			: null;
	}

	parseComment(comment: string): Comment {
		try {
			let [, type, properties] = this.parseRegex.exec(comment);
			let [style, clazz, attributes] = this.parseProperties(properties);
			return this.buildComment(type, style, clazz, attributes);
		} catch (ex) {
			console.log("ERROR: Cannot parse comment: " + comment);
			return null;
		}
	}

	buildComment(type: string, style: string[] = [], clazz: string[] = [], attributes: Map<string, string> = new Map<string, string>()): Comment {
		return {
			type: type,
			style: style,
			clazz: clazz,
			attributes: attributes,
		}
	}

	private attributeMapToString(attributes: Map<string, string>): string {
		let result = [];
		for (const [key, value] of attributes) {
			result.push(`${key}="${value}"`);
		}
		return result.join(" ");
	}

	private lineHasComment(line: string): boolean {
		return line?.includes("<!--") && line?.includes("-->");
	}

	private readCommentStringFromLine(line: string): string {
		return this.readCommentRegex.exec(line)?.[0] ?? '';
	}

	private parseProperties(properties: string): [string[], string[], Map<string, string>] {
		let result: [string[], string[], Map<string, string>] = [[], [], new Map<string, string>()];

		if (properties.includes('style=')) {
			result[0] = this.parseStyle(properties);
		}

		if (properties.includes('class=')) {
			result[1] = this.parseClass(properties);
		}

		result[2] = this.parseAttributes(properties);

		return result;
	}

	private parseStyle(comment: string): string[] {
		let result = this.styleRegex.exec(comment)?.[1] ?? '';
		return result.split(";").map((attr) => {
			return attr.trim();
		});
	}

	private parseClass(comment: string): string[] {
		let result = this.classRegex.exec(comment)?.[1] ?? '';
		return result.split(" ").map((attr) => {
			return attr.trim();
		});
	}

	private parseAttributes(properties: string): Map<string, string> {
		let m;
		const attributes = new Map<string, string>();

		while ((m = this.parsePropertiesRegex.exec(properties)) !== null) {
			if (m.index === this.parsePropertiesRegex.lastIndex) {
				this.parsePropertiesRegex.lastIndex++;
			}

			let key: string;
			let value: string;

			m.forEach((match, groupIndex) => {
				if (groupIndex == 1 || groupIndex == 3) {
					key = match;
				}
				if (groupIndex == 2 || groupIndex == 4) {
					value = match;
					attributes.set(key, value);
				}
			});

		}

		attributes.delete(undefined);
		attributes.delete('style');
		attributes.delete('class');
		return attributes;
	}

}
