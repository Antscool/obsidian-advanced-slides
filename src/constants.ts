export const ICON_DATA = '<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100"><path fill="currentColor" stroke="currentColor" d="M100 74.242V1.516H0v72.726h45.453v15.153H31.816v9.09h36.368v-9.09H54.547V74.242Zm-90.91-9.09V10.605h81.82v54.547Zm0 0"/><path fill="currentColor" stroke="currentColor" d="M54.547 19.695h9.09V56.06h-9.09ZM72.727 25.758h9.09v30.305h-9.09ZM36.363 31.816h9.09V56.06h-9.09ZM18.184 22.727h9.09v33.335h-9.09Zm0 0"/></svg>';
