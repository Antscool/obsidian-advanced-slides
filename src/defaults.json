{
	"theme": "black",
	"highlightTheme": "zenburn",
	"template": "template/reveal.html",
	"separator": "\r?\n---\r?\n",
	"verticalSeparator": "\r?\n--\r?\n",
	"enableLinks": false,
	"title": "",
	"css": ""
}
