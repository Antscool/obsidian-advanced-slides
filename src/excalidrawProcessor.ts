import { Notice } from "obsidian";
import { ObsidianUtils } from "./obsidianUtils";

export class ExcalidrawProcessor {
	private excalidrawImageRegex = /!\[\[(.*\.excalidraw)\|?([^\]]*)??\]\]\s?(<!--.*-->)?/i

	private utils: ObsidianUtils;

	constructor(utils: ObsidianUtils) {
		this.utils = utils;
	}

	process(markdown: string){
		return markdown
			.split('\n')
			.map((line) => {
				if (this.excalidrawImageRegex.test(line))
					return this.transformLine(line);
				return line;
			})
			.join('\n');
	}

	private transformLine(line: string) {

		var [, image, ext, comment] = this.excalidrawImageRegex.exec(line);
		var imgFile = this.utils.findImageEx(image);

		if (imgFile === null) {
			new Notice(`Cannot find Image for ${image}. Make sure to activate Auto-export SVG/PNG in Excalidraw Settings.`, 8000);
			return line;
		}
		return `![[${imgFile}${ext==undefined ? '' : '|' + ext}]] ${comment ?? ''}`;
	}
}


