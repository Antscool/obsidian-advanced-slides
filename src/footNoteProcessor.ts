
export class FootnoteProcessor {

	private regex = /\[\^([^\]]*)]/gmi;

	process(markdown: string, options: any){

		var output = markdown;

		var result = markdown
        .split(new RegExp(options.separator, 'gmi'))
        .map((slidegroup, index) => {
            return slidegroup
                .split(new RegExp(options.verticalSeparator, 'gmi'))
                .map((slide, index) => {
                    if (this.regex.test(slide)) {
						var newSlide = this.transformFootNotes(slide);
						output = output.replace(slide, newSlide);
                        return newSlide;
                    }
                    return slide;
                })
                .join(options.verticalSeparator);
        })
        .join(options.separator);
    return output;
	}

	transformFootNotes(markdown: string) {

		var input = markdown;
		var noteIdx = 1;
	
		var footNotes = new Map();
	
		var reResult : RegExpExecArray;
		while (reResult = this.regex.exec(input)) {
	
			input = input
				.split('\n')
				.map((line, index) => {
					if (line.includes(reResult[0])) {
						if (line.includes(reResult[0] + ": ")) {
							if (!footNotes.has(reResult[1])) {
								footNotes.set(reResult[1], line.split(reResult[0] + ": ")[1]);
							}
							return "";
						} else {
							var split = line.split(reResult[0]);
	
							var result = split[0].trim();
							result += '<sup id="fnref:' + reResult[1] + '" role="doc-noteref">' + noteIdx + '</sup>';
							result += '\n' + split[1].trim();
	
							noteIdx = noteIdx + 1;
	
							return result;
	
						}
					}
					return line;
				})
				.join('\n');
	
	
		}
	
		var footNotesBlock = '';
		footNotesBlock += '<div class="footnotes" role="doc-endnotes">\n';
		footNotesBlock += '<ol>\n';
	
		footNotes.forEach((value, key) => {
			footNotesBlock += '<li id="fn:' + key + '" role="doc-endnote"><p>' + value + '&nbsp;</p></li>';
		});
	
	
		footNotesBlock += '</ol>\n';
		footNotesBlock += '</div>\n';
	
		return input + '\n' + footNotesBlock;
	}

}


