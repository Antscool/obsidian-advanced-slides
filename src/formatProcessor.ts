
export class FormatProcessor {

	private regex = /==([^=]*)==/gm;
	private commentRegex = /%%([^%]*)%%/gm;

	process(markdown: string){
		return markdown.replaceAll('__','**')
		.replaceAll(this.regex,'<mark>$1</mark>')
		.replaceAll(this.commentRegex,'');
	}

}
