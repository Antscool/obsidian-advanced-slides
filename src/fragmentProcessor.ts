import { CommentParser } from "./comment";
import { Options } from "./options";

export class FragmentProcessor {

	private parser: CommentParser;
	private fragmentCounter = 1;
	private orderedListRegex = /\d\) /g;

	constructor() {
		this.parser = new CommentParser();
	}

	process(markdown: string, options: Options) {
		let output = markdown;

		markdown
			.split(new RegExp(options.separator, 'gmi'))
			.map((slidegroup) => {
				return slidegroup
					.split(new RegExp(options.verticalSeparator, 'gmi'))
					.map((slide) => {
						this.fragmentCounter = 1;

						const newSlide = slide.split('\n')
							.map((line) => {
								if (line.trim().startsWith("+ ") || this.orderedListRegex.test(line.trim())) {
									return this.transformLine(line);
								}
								return line;
							})
							.join('\n');
						output = output.split(slide).join(newSlide);
						return newSlide;

					})
					.join(options.verticalSeparator);
			})
			.join(options.separator);

		return output;
	}

	transformLine(line: string) {
		const comment = this.parser.parseLine(line) ?? this.parser.buildComment('element');

		if (line.includes('<!--')) {
			line = line.substring(0, line.indexOf('<!--'));
		}

		line = line.replaceAll("+ ", "- ");
		line = line.replaceAll(this.orderedListRegex, '1. ');

		if (!comment.attributes.get('data-fragment-index')) {
			comment.attributes.set("data-fragment-index", this.fragmentCounter.toString());
			if (!comment.clazz.includes('fragment')) {
				comment.clazz.push('fragment');
			}
			this.fragmentCounter++;
		}
		const output = line + this.parser.commentToString(comment);
		return output;
	}

}
