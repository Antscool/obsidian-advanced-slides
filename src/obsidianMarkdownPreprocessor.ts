import { App } from "obsidian";
import { BlockProcessor } from "./blockProcessor";
import { FootnoteProcessor } from "./footNoteProcessor";
import { FormatProcessor } from "./formatProcessor";
import { ImageProcessor } from "./imageProcessor";
import { InternalLinkProcessor } from "./internalLinkProcessor";
import { LatexProcessor } from "./latexProcessor";
import { MultipleFileProcessor } from "./multipleFileProcessor";

export class ObsidianMarkdownPreprocessor {

	private multipleFileProcessor : MultipleFileProcessor;
	private blockProcessor : BlockProcessor;
	private imageProcessor : ImageProcessor;
	private internalLinkProcessor : InternalLinkProcessor;
	private footnoteProcessor : FootnoteProcessor;
	private latexProcessor : LatexProcessor;
	private formatProcessor : FormatProcessor;

	constructor(app: App) {
		this.multipleFileProcessor = new MultipleFileProcessor(app);
		this.blockProcessor = new BlockProcessor();
		this.imageProcessor = new ImageProcessor(app);
		this.internalLinkProcessor = new InternalLinkProcessor();
		this.footnoteProcessor = new FootnoteProcessor();
		this.latexProcessor = new LatexProcessor();
		this.formatProcessor = new FormatProcessor();
	}

	process(markdown: string, options: any){
		const afterMultipleFileProcessor = this.multipleFileProcessor.process(markdown);
		const afterBlockProcessor = this.blockProcessor.process(afterMultipleFileProcessor);
		const afterFootNoteProcessor = this.footnoteProcessor.process(afterBlockProcessor, options);
		const afterImageProcessor = this.imageProcessor.process(afterFootNoteProcessor);
		const afterInternalLinkProcessor = this.internalLinkProcessor.process(afterImageProcessor);
		const afterLatexProcessor = this.latexProcessor.process(afterInternalLinkProcessor);
		const afterFormatProcessor = this.formatProcessor.process(afterLatexProcessor);
		return afterFormatProcessor;
	}



}


