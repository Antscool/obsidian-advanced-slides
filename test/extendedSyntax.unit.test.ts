import { ObsidianMarkdownPreprocessor } from "src/obsidianMarkdownPreprocessor";
import { prepare } from "./testUtils";
import { MockedObsidianUtils, obsidianUtils as utilsInstance} from "./__mocks__/mockObsidianUtils";


test('Extended Markdown Syntax > Horizontal / Vertical Slides', () => {

	const input =
`# Slide 1

---

# Slide 2.1

--

# Slide 2.2
`;

	const { options, markdown } = prepare(input);
	var sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Extended Markdown Syntax >  Element Annotations', () => {

	const input =
`text with border <!-- .element: class="with-border" -->

text with background <!-- .element: style="background:blue" -->

text with attribute <!-- .element: data-toggle="modal" -->
`;

	const { options, markdown } = prepare(input);
	var sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Extended Markdown Syntax >  Slide Annotations', () => {

	const input =
`<!-- .slide: style="background-color: coral;" -->

# Header with coral background color

Paragraph has coral background color, too!

---

<!-- .slide: style="background-color: green;" -->

- All Bullet points
- have green
- background color
`;

	const { options, markdown } = prepare(input);
	var sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Extended Markdown Syntax >  Block Comments', () => {

	const input =
`::: block

#### Header
_and_
Paragraph content
*in same block*

:::

---

no color

::: block <!-- .element: style="background-color: red;" -->

everything inside this block has red background color

::: block <!-- .element: style="background-color: blue;" -->

blue

:::

red

:::

no color
`;

	const { options, markdown } = prepare(input);
	var sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Extended Markdown Syntax >  Fragments', () => {

	const input =
`Fade in <!-- .element: class="fragment" -->

Fade out <!-- .element: class="fragment fade-out" -->

Highlight red <!-- .element: class="fragment highlight-red" -->

Fade in, then out <!-- .element: class="fragment fade-in-then-out" -->

Slide up while fading in <!-- .element: class="fragment fade-up" -->

---

- Permanent item
- Appear Fourth <!-- .element: class="fragment" data-fragment-index="4" -->
- Appear Third <!-- .element: class="fragment" data-fragment-index="3" -->
- Appear Second <!-- .element: class="fragment" data-fragment-index="2" -->
- Appear First <!-- .element: class="fragment" data-fragment-index="1" -->
`;

	const { options, markdown } = prepare(input);
	var sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Extended Markdown Syntax >  Inline Styling', () => {

	const input =
`<style>
.with-border{
	border: 1px solid red;
}
</style>

styled text <!-- .element: class="with-border" -->
`;

	const { options, markdown } = prepare(input);
	var sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Extended Markdown Syntax >  Slide Backgrounds', () => {

	const input =
`<!-- .slide: data-background="aquamarine" -->
## Slide with text based background
---

<!-- .slide: data-background="#ff0000" -->
## Slide with hex based background

---

<!-- .slide: data-background="rgb(70, 70, 255)" -->
## Slide with rgb based background

---

<!-- .slide: data-background="hsla(315, 100%, 50%, 1)" -->
## Slide with hsl based background

---

# Slide without background

---

<!-- .slide: data-background-image="https://picsum.photos/seed/picsum/800/600" -->
## Slide with image background

---

<!-- .slide: data-background-image="https://picsum.photos/seed/picsum/800/600" data-background-opacity="0.5" -->
## with opacity

0.5 â‰™ 50% opacity

---

## More options:

See [reveal backgrounds](https://revealjs.com/backgrounds/)
`;

	const { options, markdown } = prepare(input);
	var sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Extended Markdown Syntax >  Speaker Notes', () => {

	const input =
`## My Slide

This is part of my Presentation


note: this is not! Only the speaker might see this text.

- and this bulletpoint
- or this picture

![](https://picsum.photos/id/1005/250/250) 
`;

	const { options, markdown } = prepare(input);
	var sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Extended Markdown Syntax >  Fragmented list', () => {

	const input =
`# Unordered list

- First
- Second
- Third

---

# Fragmented unordered list

- Permanent
+ First
+ Second
+ Third

---

# Ordered list

1. First
2. Second
3. Third

---

# Fragmented ordered list

1. Permanent
2) Second
3) Third
4) Fourth
`;

	const { options, markdown } = prepare(input);
	var sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});
