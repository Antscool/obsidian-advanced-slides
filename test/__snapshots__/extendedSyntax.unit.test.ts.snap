// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advanced Slides Feature >  Show Debug Grid 1`] = `
"## My Slide

This slide shows the debug view feature


<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 0px; top: 0px; height: 70px; min-height: 70px; width: 960px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 0px; top: 70px; height: 70px; min-height: 70px; width: 960px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 0px; top: 140px; height: 70px; min-height: 70px; width: 960px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 0px; top: 210px; height: 70px; min-height: 70px; width: 960px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 0px; top: 280px; height: 70px; min-height: 70px; width: 960px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 0px; top: 350px; height: 70px; min-height: 70px; width: 960px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 0px; top: 420px; height: 70px; min-height: 70px; width: 960px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 0px; top: 490px; height: 70px; min-height: 70px; width: 960px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 0px; top: 560px; height: 70px; min-height: 70px; width: 960px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 0px; top: 630px; height: 70px; min-height: 70px; width: 960px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 0px; top: 0px; height: 700px; min-height: 700px; width: 96px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 96px; top: 0px; height: 700px; min-height: 700px; width: 96px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 192px; top: 0px; height: 700px; min-height: 700px; width: 96px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 288px; top: 0px; height: 700px; min-height: 700px; width: 96px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 384px; top: 0px; height: 700px; min-height: 700px; width: 96px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 480px; top: 0px; height: 700px; min-height: 700px; width: 96px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 576px; top: 0px; height: 700px; min-height: 700px; width: 96px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 672px; top: 0px; height: 700px; min-height: 700px; width: 96px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 768px; top: 0px; height: 700px; min-height: 700px; width: 96px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>
<div class=\\"reset-margin\\" style=\\"border: thin dotted blue; box-sizing: border-box; position: absolute; left: 864px; top: 0px; height: 700px; min-height: 700px; width: 96px; display: flex; flex-direction: column; align-items: center; justify-content: space-evenly\\" >
</div>

<!-- .slide: class=\\"drop\\" -->
"
`;

exports[`Extended Markdown Syntax >  Block Comments 1`] = `
"<div class=\\"block\\">

#### Header
_and_
Paragraph content
*in same block*

</div>



---

no color

<div class=\\"block\\">

<!-- .element: style=\\"background-color: red\\" -->

everything inside this block has red background color

<div class=\\"block\\">

<!-- .element: style=\\"background-color: blue\\" -->

blue

</div>



red

</div>



no color
"
`;

exports[`Extended Markdown Syntax >  Chart support > Extended Settings 1`] = `
"
<canvas style=\\"max-height:700px\\" data-chart=\\"bar\\" >
<!--
{\\"data\\":{\\"datasets\\":[{\\"data\\":[1,2,3,4,5,6,7,8,9],\\"label\\":\\"Title 1\\",\\"backgroundColor\\":\\"#4285f4\\"},{\\"data\\":[5,4,3,2,1,0,-1,-2,-3],\\"label\\":\\"Title 2\\",\\"backgroundColor\\":\\"#ea4335\\"}],\\"labels\\":[\\"Monday\\",\\"Tuesday\\",\\"Wednesday\\",\\"Thursday\\",\\"Friday\\",\\"Saturday\\",\\"Sunday\\",\\"next Week\\",\\"next Month\\"]},\\"options\\":{\\"elements\\":{\\"bar\\":{\\"spanGaps\\":true,\\"tension\\":0.5}},\\"scales\\":{\\"y\\":{\\"beginAtZero\\":true,\\"stacked\\":true},\\"x\\":{\\"stacked\\":true}},\\"plugins\\":{\\"legend\\":{\\"display\\":false,\\"position\\":\\"left\\"}}}}-->
</canvas>
"
`;

exports[`Extended Markdown Syntax >  Chart support > Illegal Input 1`] = `
"
	
	\`\`\`chart
    type: bar
    labels: [Monday,Tuesday,Wednesday,Thursday,Friday, Saturday, Sunday, \\"next Week\\", \\"next Month\\"]
    series:
      - title: Title 1
        data: [1,2,3,4,5,6,7,8,9]
      - title: Title 2
        data: [5,4,3,2,1,0,-1,-2,-3]
\`

	"
`;

exports[`Extended Markdown Syntax >  Chart support 1`] = `
"
<canvas style=\\"max-height:700px\\" data-chart=\\"bar\\" >
<!--
{\\"data\\":{\\"datasets\\":[{\\"data\\":[1,2,3,4,5,6,7,8,9],\\"label\\":\\"Title 1\\",\\"backgroundColor\\":\\"#4285f4\\"},{\\"data\\":[5,4,3,2,1,0,-1,-2,-3],\\"label\\":\\"Title 2\\",\\"backgroundColor\\":\\"#ea4335\\"}],\\"labels\\":[\\"Monday\\",\\"Tuesday\\",\\"Wednesday\\",\\"Thursday\\",\\"Friday\\",\\"Saturday\\",\\"Sunday\\",\\"next Week\\",\\"next Month\\"]},\\"options\\":{\\"elements\\":{\\"bar\\":{}}}}-->
</canvas>
"
`;

exports[`Extended Markdown Syntax >  Element Annotations 1`] = `
"text with border <!-- .element: class=\\"with-border\\" -->

text with background <!-- .element: style=\\"background: blue\\" -->

text with attribute <!-- .element: data-toggle=\\"modal\\" -->
"
`;

exports[`Extended Markdown Syntax >  Excalidraw support 1`] = `
"#### Excalidraw support

<p style=\\"line-height: 0\\" class=\\"reset-paragraph image-paragraph\\"><img src=\\"path/to/Sample.excalidraw.svg\\" alt=\\"\\" style=\\"width: 100px\\"></img></p>


<p style=\\"line-height: 0\\" class=\\"reset-paragraph image-paragraph\\"><img src=\\"path/to/Sample.excalidraw.svg\\" alt=\\"\\" style=\\"width: 300px; height: 400px\\"></img></p>
"
`;

exports[`Extended Markdown Syntax >  Fragmented list 1`] = `
"# Unordered list

- First
- Second
- Third

---

# Fragmented unordered list

- Permanent
- &shy;<!-- .element: class=\\"fragment\\" data-fragment-index=\\"1\\" -->First
- &shy;<!-- .element: class=\\"fragment\\" data-fragment-index=\\"2\\" -->Second
- &shy;<!-- .element: class=\\"fragment\\" data-fragment-index=\\"3\\" -->Third

---

# Fragmented unordered list with ignored code block

- First
- &shy;<!-- .element: class=\\"fragment\\" data-fragment-index=\\"1\\" -->Second

\`\`\`diff
- ignored block text 1
+ ignored block text 2
\`\`\`

---

# Ordered list

1. First
2. Second
3. Third

---

# Fragmented ordered list

1. Permanent
1. &shy;<!-- .element: class=\\"fragment\\" data-fragment-index=\\"1\\" -->Second
1. &shy;<!-- .element: class=\\"fragment\\" data-fragment-index=\\"2\\" -->Third
1. &shy;<!-- .element: class=\\"fragment\\" data-fragment-index=\\"3\\" -->Fourth
"
`;

exports[`Extended Markdown Syntax >  Fragments 1`] = `
"Fade in <!-- .element: class=\\"fragment\\" -->

Fade out <!-- .element: class=\\"fragment fade-out\\" -->

Highlight red <!-- .element: class=\\"fragment highlight-red\\" -->

Fade in, then out <!-- .element: class=\\"fragment fade-in-then-out\\" -->

Slide up while fading in <!-- .element: class=\\"fragment fade-up\\" -->

---

- Permanent item
- Appear Fourth <!-- .element: class=\\"fragment\\" data-fragment-index=\\"4\\" -->
- Appear Third <!-- .element: class=\\"fragment\\" data-fragment-index=\\"3\\" -->
- Appear Second <!-- .element: class=\\"fragment\\" data-fragment-index=\\"2\\" -->
- Appear First <!-- .element: class=\\"fragment\\" data-fragment-index=\\"1\\" -->
"
`;

exports[`Extended Markdown Syntax >  Inline Styling 1`] = `
"<style>
.with-border{
	border: 1px solid red;
}
</style>

styled text <!-- .element: class=\\"with-border\\" -->
"
`;

exports[`Extended Markdown Syntax >  Slide Annotations 1`] = `
"<!-- .slide: style=\\"background-color: coral\\" -->

# Header with coral background color

Paragraph has coral background color, too!

---

<!-- .slide: style=\\"background-color: green\\" -->

- All Bullet points
- have green
- background color
"
`;

exports[`Extended Markdown Syntax >  Slide Backgrounds 1`] = `
"<!-- .slide: data-background=\\"aquamarine\\" -->
## Slide with text based background
---

<!-- .slide: data-background=\\"#ff0000\\" -->
## Slide with hex based background

---

<!-- .slide: data-background=\\"rgb(70, 70, 255)\\" -->
## Slide with rgb based background

---

<!-- .slide: data-background=\\"hsla(315, 100%, 50%, 1)\\" -->
## Slide with hsl based background

---

# Slide without background

---

<!-- .slide: data-background-image=\\"https://picsum.photos/seed/picsum/800/600\\" -->
## Slide with image background

---

<!-- .slide: data-background-image=\\"/documentation/Image.jpg\\" -->
## Slide with image background #2

---

<!-- .slide: data-background-image=\\"https://picsum.photos/seed/picsum/800/600\\" data-background-opacity=\\"0.5\\" -->
## with opacity

0.5 ≙ 50% opacity

---

## More options:

See [reveal backgrounds](https://revealjs.com/backgrounds/)
"
`;

exports[`Extended Markdown Syntax >  Speaker Notes 1`] = `
"## My Slide

This is part of my Presentation


note: this is not! Only the speaker might see this text.

- and this bulletpoint
- or this picture

<p style=\\"line-height: 0\\" class=\\"reset-paragraph image-paragraph\\"><img src=\\"https://picsum.photos/id/1005/250/250\\" alt=\\"\\" ></img></p>

"
`;

exports[`Extended Markdown Syntax > Horizontal / Vertical Slides 1`] = `
"# Slide 1

---

# Slide 2.1

--

# Slide 2.2
"
`;
